pkg/plugin/driver/job/job_test.go

```diff
@@ -160,3 +166,77 @@ func TestFillTemplate(t *testing.T) {
 	}
 
 }
+
+func TestMonitorOnce(t *testing.T) {
+	testCases := []struct {
+		desc       string
+		expectDone bool
+
+		// Ensure we are getting the err result we expect; lots of ways to get errors
+		// that may not be clear.
+		expectErrResultMsg string
+
+		job           *Plugin
+		podOnServer   *corev1.Pod
+		errFromServer error
+	}{
+		{
+			desc:       "Cleaned up indicates exit without error",
+			expectDone: true,
+			job:        &Plugin{driver.Base{CleanedUp: true}},
+		}, {
+			desc:               "Missing pod results in error",
+			job:                &Plugin{},
+			podOnServer:        nil,
+			errFromServer:      errors.New("forcedError"),
+			expectErrResultMsg: "forcedError",
+			expectDone:         true,
+		}, {
+			desc: "Failing pod results in error",
+			job:  &Plugin{},
+			podOnServer: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{Reason: "Unschedulable"},
+					},
+				},
+			},
+			expectErrResultMsg: "Can't schedule pod:",
+			expectDone:         true,
+		}, {
+			desc: "Healthy pod results in no error and continued monitoring",
+			job:  &Plugin{driver.Base{Namespace: "testNS"}},
+			podOnServer: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
```

Comment:
By the way, this is an important line. Without it, our mock client gets all the items, but then our sonobuoy logic is filtering out things without this label. It is important enough (and not clear) so I'll add a comment for this too.,